### PR TITLE
DT-1780 Implementation of update court details with feature switch

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -129,7 +129,8 @@ workflows:
             container=community-api,name=FEATURES_APPLY_LIMITED_ACCESS_MARKERS,                       value=true,
             container=community-api,name=FEATURES_NOMS_UPDATE_MULTIPLE_EVENTS_UPDATE_BULK_KEY_DATES,  value=false,
             container=community-api,name=FEATURES_NOMS_UPDATE_MULTIPLE_EVENTS_UPDATE_KEY_DATES,       value=true,
-            container=community-api,name=FEATURES_NOMS_UPDATE_MULTIPLE_EVENTS_UPDATE_PRISON_LOCATION, value=true
+            container=community-api,name=FEATURES_NOMS_UPDATE_MULTIPLE_EVENTS_UPDATE_PRISON_LOCATION, value=true,
+            container=community-api,name=FEATURES_REGISTERS_COURT_CODE_ALLOWED_PATTERN,               value=XXXX[A-Z]{2}
```
Comment (andymarke) [inline .circleci/config.yml:133]: For now only allow a very restricted number of courts to be updated in test, and none can be updated anywhere else

File: src/main/java/uk/gov/justice/digital/delius/service/CourtService.java [hunk 2]
```diff
@@ -14,19 +17,46 @@
...
+            courtEntity.setCounty(court.getCounty());
+            courtEntity.setFax(court.getFax());
+            courtEntity.setLocality(court.getLocality());
+            courtEntity.setPostcode(court.getPostcode());
+            courtEntity.setStreet(court.getStreet());
+            courtEntity.setTelephoneNumber(court.getTelephoneNumber());
+            courtEntity.setTown(court.getTown());
+            courtEntity.setSelectable(court.isActive() ? "Y" : "N");
+        } else {
+            log.warn(String.format("This Court Update feature for %s is currently switched off", code));
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/service/CourtService.java:44]: odd to do a `format` inside a `log.warn`

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/digital/delius/service/CourtService.java (hunk 2 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/service/CourtService.java:44]: yeah, I ALWAYS forget

**Pull request merged**